platform/linux-generic/arch/aarch64/odp_llsc.h

```diff
@@ -1,170 +0,0 @@
-/* Copyright (c) 2017, ARM Limited. All rights reserved.
- *
- * Copyright (c) 2017-2018, Linaro Limited
- * All rights reserved.
- *
- * SPDX-License-Identifier: BSD-3-Clause
- */
-
-#ifndef PLATFORM_LINUXGENERIC_ARCH_ARM_LLSC_H
-#define PLATFORM_LINUXGENERIC_ARCH_ARM_LLSC_H
-
-#ifndef PLATFORM_LINUXGENERIC_ARCH_ARM_ODP_CPU_H
-#error This file should not be included directly, please include odp_cpu.h
-#endif
-
-#include <odp_types_internal.h>
-
-static inline uint16_t ll8(uint8_t *var, int mm)
-{
-	uint16_t old;
-
-	if (mm == __ATOMIC_ACQUIRE)
-		__asm__ volatile("ldaxrb %w0, [%1]"
-				 : "=&r" (old)
-				 : "r" (var)
-				 : "memory");
-	else if (mm == __ATOMIC_RELAXED)
-		__asm__ volatile("ldxrb %w0, [%1]"
-				 : "=&r" (old)
-				 : "r" (var)
-				 : );
-	else
-		_ODP_ABORT();
-	return old;
-}
-
-static inline uint32_t ll32(uint32_t *var, int mm)
-{
-	uint32_t old;
-
-	if (mm == __ATOMIC_ACQUIRE)
-		__asm__ volatile("ldaxr %w0, [%1]"
-				 : "=&r" (old)
-				 : "r" (var)
-				 : "memory");
-	else if (mm == __ATOMIC_RELAXED)
-		__asm__ volatile("ldxr %w0, [%1]"
-				 : "=&r" (old)
-				 : "r" (var)
-				 : );
-	else
-		_ODP_ABORT();
-	return old;
-}
-
-/* Return 0 on success, 1 on failure */
-static inline uint32_t sc32(uint32_t *var, uint32_t neu, int mm)
```

Comment:
I am not sure if removal of the sc functions, even if unused now, is good, since the ll functions stay and the sc functions pair with them. Without sc, ll is can not be used for much else than with wfe.